### PR TITLE
gall: remove pending messages from breached ship

### PR DESCRIPTION
**Pull request by philipcmonk**
See #2485 for context.

When a ship breaches, we remove all messages that have yet to be
delivered to an app (eg if it's not yet started).  We also add
|gall-sear to do this manually, but this shouldn't be needed in normal
operation.

Finally, to unblock ~zod and ~bus on mainnet, we sear one particular
ship automatically on loading hood.  It cannot be done manually because
no userpace changes can be made until it's unblocked.

Tested on a localhost copy of ~bus.

### CODE REVIEW

File: pkg/arvo/sys/vane/gall.hoon [hunk 1]
```diff
@@ -724,6 +725,30 @@
...
+    |-  ^+  mo-core
+    ?:  =(~ blocked.i.agents)
+      ?~  new-blocked
+        ^$(agents t.agents)
+      %=  ^$
+        agents      t.agents
+        new-agents  (~(put by new-agents) name.i.agents new-blocked)
+      ==
+    =^  mov  blocked.i.agents  ~(get to blocked.i.agents)
+    ~!  mov
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/gall.hoon]: Do we still need this `~!` ?  Looks like a development artifact..

Comment (philipcmonk) [reply pkg/arvo/sys/vane/gall.hoon]: Good catch, also the following ~&

**Comment by philipcmonk** [comment]
I don't understand the CI failure: "The job exceeded the maximum log length, and has been terminated."  I also can't find how to restart CI.  Is this some cache miss that'll be fine on master?

**Comment by joemfb** [comment]
It looks like cachix is down or failing, so we're building *everything*. /cc @benjamin-tlon

**Comment by benjamin-tlon** [comment]
Interesting.

That's no good at all!

**Comment by benjamin-tlon** [comment]
```
$ cachix use urbit2
cachix:
  (FailureResponse
    (Request
      { requestPath =
          (BaseUrl
            { baseUrlScheme = Https
            , baseUrlHost = "cachix.org"
            , baseUrlPort = 443
            , baseUrlPath = ""
            }
          , "/api/v1/cache/urbit2"
          )
      , requestQueryString = fromList []
      , requestBody = Nothing
      , requestAccept = fromList [application/json;charset=utf-8,application/json]
      , requestHeaders = fromList [("Authorization","<REDACTED>")])
      , requestHttpVersion = HTTP/1.1
      , requestMethod = "GET"
      }
      (Response
        { responseStatusCode = Status {statusCode = 521, statusMessage = ""}
        , responseHeaders = fromList
            [ ("Date","Thu, 19 Mar 2020 18:38:31 GMT")
            , ("Content-Type","text/html; charset=UTF-8")
            , ("Transfer-Encoding","chunked")
            , ("Connection","keep-alive")
            , ("Set-Cookie","cf_ob_info=521:57694f939965c508:ORD; path=/; expires=Thu, 19-Mar-20 18:39:01 GMT")
            , ("Set-Cookie","cf_use_ob=443; path=/; expires=Thu, 19-Mar-20 18:39:01 GMT")
            , ("Expires","Thu, 01 Jan 1970 00:00:01 GMT")
            , ("Cache-Control","no-store, no-cache, must-revalidate, post-check=0, pre-check=0")
            , ("Pragma","no-cache")
            , ("X-Frame-Options","SAMEORIGIN")
            , ("CF-RAY","57694f939965c508-ORD")
            , ("Server","cloudflare")
            ]
        , responseHttpVersion = HTTP/1.1
        , responseBody = "<!DOCTYPE html>\n<!--[if lt IE 7]> <html class=\"no-js ie6 oldie\" lang=\"en-US\"> <![endif]-->\n<!--[if IE 7]>    <html class=\"no-js ie7 oldie\" lang=\"en-US\"> <![endif]-->\n<!--[if IE 8]>    <html class=\"no-js ie8 oldie\" lang=\"en-US\"> <![endif]-->\n<!--[if gt IE 8]><!--> <html class=\"no-js\" lang=\"en-US\"> <!--<![endif]-->\n<head>\n<meta http-equiv=\"refresh\" content=\"0\">\n\n<title>cachix.org | 521: Web server is down</title>\n<meta charset=\"UTF-8\" />\n<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n<meta http-equiv=\"X-UA-Compatible\" content=\"IE=Edge,chrome=1\" />\n<meta name=\"robots\" content=\"noindex, nofollow\" />\n<meta name=\"viewport\" content=\"width=device-width,initial-scale=1,maximum-scale=1\" />\n<link rel=\"stylesheet\" id=\"cf_styles-css\" href=\"/cdn-cgi/styles/cf.errors.css\" type=\"text/css\" media=\"screen,projection\" />\n<!--[if lt IE 9]><link rel=\"stylesheet\" id='cf_styles-ie-css' href=\"/cdn-cgi/styles/cf.errors.ie.css\" type=\"text/css\" media=\"screen,projection\" /><![endif]-->\n<style type=\"text/css\">body{margin:0;padding:0}</style>\n\n\n\n\n</head>\n<body>\n<div id=\"cf-wrapper\">\n\n    \n\n    <div id=\"cf-error-details\" class=\"cf-error-details-wrapper\">\n        <div class=\"cf-wrapper cf-error-overview\">\n            <h1>\n              \n              <span class=\"cf-error-type\">Error</span>\n              <span class=\"cf-error-code\">521</span>\n              <small class=\"heading-ray-id\">Ray ID: 57694f939965c508 &bull; 2020-03-19 18:38:31 UTC</small>\n            </h1>\n            <h2 class=\"cf-subheadline\">Web server is down</h2>\n        </div><!-- /.error-overview -->\n        \n        <div class=\"cf-section cf-highlight cf-status-display\">\n            <div class=\"cf-wrapper\">\n                <div class=\"cf-columns cols-3\">\n                  \n<div id=\"cf-browser-status\" class=\"cf-column cf-status-item cf-browser-status \">\n  <div class=\"cf-icon-error-container\">\n    <i class=\"cf-icon cf-icon-browser\"></i>\n    <i class=\"cf-icon-status cf-icon-ok\"></i>\n  </div>\n  <span class=\"cf-status-desc\">You</span>\n  <h3 class=\"cf-status-name\">Browser</h3>\n  <span class=\"cf-status-label\">Working</span>\n</div>\n\n<div id=\"cf-cloudflare-status\" class=\"cf-column cf-status-item cf-cloudflare-status \">\n  <div class=\"cf-icon-error-container\">\n    <i class=\"cf-icon cf-icon-cloud\"></i>\n    <i class=\"cf-icon-status cf-icon-ok\"></i>\n  </div>\n  <span class=\"cf-status-desc\">Chicago</span>\n  <h3 class=\"cf-status-name\">Cloudflare</h3>\n  <span class=\"cf-status-label\">Working</span>\n</div>\n\n<div id=\"cf-host-status\" class=\"cf-column cf-status-item cf-host-status cf-error-source\">\n  <div class=\"cf-icon-error-container\">\n    <i class=\"cf-icon cf-icon-server\"></i>\n    <i class=\"cf-icon-status cf-icon-error\"></i>\n  </div>\n  <span class=\"cf-status-desc\">cachix.org</span>\n  <h3 class=\"cf-status-name\">Host</h3>\n  <span class=\"cf-status-label\">Error</span>\n</div>\n\n                </div>\n              \n            </div>\n        </div><!-- /.status-display -->\n\n        <div class=\"cf-section cf-wrapper\">\n            <div class=\"cf-columns two\">\n                <div class=\"cf-column\">\n                    <h2>What happened?</h2>\n                    <p>The web server is not returning a connection. As a result, the web page is not displaying.</p>\n                </div>\n              \n                <div class=\"cf-column\">\n                    <h2>What can I do?</h2>\n                          <h5>If you are a visitor of this website:</h5>\n      <p>Please try again in a few minutes.</p>\n\n      <h5>If you are the owner of this website:</h5>\n      <p><span>Contact your hosting provider letting them know your web server is not responding.</span> <a href=\"https://support.cloudflare.com/hc/en-us/articles/200171916-Error-521\">Additional troubleshooting information</a>.</p>\n                </div>\n            </div>\n              \n        </div><!-- /.section -->\n\n        <div class=\"cf-error-footer cf-wrapper\">\n  <p>\n    <span class=\"cf-footer-item\">Cloudflare Ray ID: <strong>57694f939965c508</strong></span>\n    <span class=\"cf-footer-separator\">&bull;</span>\n    <span class=\"cf-footer-item\"><span>Your IP</span>: 34.66.25.221</span>\n    <span class=\"cf-footer-separator\">&bull;</span>\n    <span class=\"cf-footer-item\"><span>Performance &amp; security by</span> <a href=\"https://www.cloudflare.com/5xx-error-landing?utm_source=error_footer\" id=\"brand_link\" target=\"_blank\">Cloudflare</a></span>\n    \n  </p>\n</div><!-- /.error-footer -->\n\n\n    </div><!-- /#cf-error-details -->\n</div><!-- /#cf-wrapper -->\n</body>\n</html>\n\n"
        }))
```

**Comment by benjamin-tlon** [comment]
```
<h2 class=\"cf-subheadline\">Web server is down</h2>
```

**Comment by benjamin-tlon** [comment]
I'm not sure why the OSX build is failing too.

**Comment by benjamin-tlon** [comment]
That one worked (CI), looks like cachix had some downtime.

**Comment by philipcmonk** [comment]
Except the second part is listed as pending forever.

**Comment by loganallenc** [comment]
@ixv this can be merged

**Comment by ixv** [comment]
There is a conflict.

**Comment by philipcmonk** [comment]
Only in the pill, right?

**Pull request merged**